phpBB/includes/acp/acp_bbcodes.php

```diff
@@ -232,6 +238,11 @@ function main($id, $mode)
 						trigger_error($user->lang['BBCODE_HELPLINE_TOO_LONG'] . adm_back_link($this->u_action), E_USER_WARNING);
 					}
 
+					if (strlen($bbcode_font_icon) > 64 && preg_match('/^[\w-]+$/', $bbcode_font_icon))
```

Comment:
```suggestion
					if (strlen($bbcode_font_icon) > 64 || preg_match('/^[\w-]+$/', $bbcode_font_icon))
```
Would make more sense with the suggested change.